netket/sampler/metropolis.py

```diff
@@ -262,6 +264,33 @@ def __post_init__(self):
         if self.n_sweeps is None:
             object.__setattr__(self, "n_sweeps", self.hilbert.size)
 
+    def sample_next(
+        sampler,
+        machine: Union[Callable, nn.Module],
+        parameters: PyTree,
+        state: Optional[SamplerState] = None,
+    ) -> Tuple[jnp.ndarray, SamplerState]:
+        """
+        Samples the next state in the markov chain.
```

Comment:
```suggestion
        Samples the next state in the markov chain.
        
        If you are defining your own Metropolis sampler, don't override this but override `_sample_next` instead.
```

Reply:
You should also add this docstring to `_sample_next` because it's anymore in the base sampler. 
```
        """
        Implementation of sample_next for subclasses of Sampler.
        If you sub-class Sampler, you should define _sample_next and not sample_next
        itself, because reset contains some common logic.
        """
```